gapic/src/main/com/google/gapid/perfetto/models/FrameInfo.java

```diff
@@ -31,70 +37,307 @@
 public class FrameInfo {
   public static final FrameInfo NONE = new FrameInfo(Collections.emptyList());
 
-  private static final String FRAME_SLICE_QUERY =
-      "select t.id, t.name, t.scope, max(depth) + 1 " +
-      "from gpu_track t join frame_slice s on (t.id = s.track_id) " +
-      "group by t.id " +
-      "order by t.id";
-  private static final String DISPLAYED_FRAME_TRACK_NAME = "Displayed Frame";
+  private static final String LAYER_NAME_QUERY =
+      "select distinct layer_name from frame_slice";
+  private static final String LAYER_TRACKS_QUERY =
+      "select name, track_id from gpu_track inner join " +
+      "(select distinct track_id from frame_slice where layer_name = '%s') t" +
+      " on gpu_track.id = t.track_id";
+  private static final String TRACK_ID_QUERY =
+      "select group_concat(track_id) from (%s) where name GLOB '%s'";
 
-  private final List<Buffer> buffers;
+  // The following queries use experimental_slice_layout from perfetto. Its a function which takes
+  // comma separated string of track_ids and stacks overlapping slices vertically. This is useful
+  // for generating ownership phases where there can be more than one slice (not related) at the
+  // same time.
 
-  private FrameInfo(List<Buffer> buffers) {
-    this.buffers = buffers;
-  }
+  // Cast name as int for the phases. This is needed because experimental_slice_layout generates
+  // columns only based on slice_table. frame_slice is a child of slice_table and the extra columns
+  // such as frame_number and layer_name won't be generated. The alternate is doing a join on
+  // frame_slice table with slice_id but that query is very expensive in that it visibly slows down
+  // the loading of tracks.
```

Comment:
Ugh....we need to figure out how to fix this.  This is a recipe for disaster.  Someone is going to change the name some day and break things because they did not know/forgot the name has to be the frame number

Reply:
Added a TODO for this.